module-bsp/board/rt1051/puretx/hal/battery_charger/BatteryCharger.cpp

```diff
@@ -1,120 +1,201 @@
 // Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
-#include "BatteryCharger.hpp"
+#include "BatteryChargerIRQ.hpp"
```

Comment:
All battery charging has to be retested thoroughly after such a changes. You have to be aware that some things can be broken since this code is outcome of many trials and errors.

Reply:
Yeah, I'm aware of this. That's why I also tested it on Pure ;)

Reply:
Do you have any specific test cases you would like me to test? So far I tested:

- brown-out notifications, 
- I'm in the middle of testing the full charge, then full discharge with the phone switched on
- I also discharged the battery completely while in bootloader mode and then proceeded to launch the phone(with and without a charger plugged in)
- I also tested battery critical-level notifications(with and without charger plugged in)

Reply:
I'll try to write down everything what comes to my mind:
- Charging until fully charged (it should communicate pluggedNotCharging) 
- Charging current changes due to different types of USB source
- Charging control due to temperature -> heating with hairdryer or cooling battery with ice is helpful.
- Proper readout of the state at system boot (there were some non-deterministic issues after many trials)
- Restart of the system with USB plugged -> charger is constantly on